pkg/virtctl/vm/vm.go

```diff
@@ -389,9 +429,85 @@ func getDryRunOption(dryRun bool) []string {
 	return nil
 }
 
-func (o *Command) Run(args []string) error {
+func validateOutputFormat() error {
+	if outputFormat != YAML && outputFormat != JSON {
+		return fmt.Errorf("error not supported output format defined: %+v", outputFormat)
+	}
+
+	return nil
+}
+
+func readVMFromFile(filePath string) (*v1.VirtualMachine, error) {
+	vm := &v1.VirtualMachine{}
+
+	readFile, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading file %+v", err)
+	}
+
+	err = yaml.Unmarshal(readFile, vm)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding Virtual Machine %+v", err)
+	}
+
+	return vm, nil
+}
+
+func printExpandedVM(outputFormat string, expandedVm *v1.VirtualMachine) error {
+	var formatedOutput []byte
+	var err error
+
+	switch outputFormat {
+	case JSON:
+		formatedOutput, err = json.MarshalIndent(expandedVm, "", " ")
+	case YAML:
+		formatedOutput, err = yaml.Marshal(expandedVm)
+	}
+
+	if err != nil {
+		return err
+	}
 
-	vmiName := args[0]
+	fmt.Printf("%+v", string(formatedOutput))
+	return nil
+}
+
+func expandVirtualMachine(virtClient kubecli.KubevirtClient, namespace string, args []string) error {
+	var expandedVm *v1.VirtualMachine
+	var err error
+
+	if len(args) == 0 && filePath != "" {
+		vm, err := readVMFromFile(filePath)
+		if err != nil {
+			return err
+		}
+
+		expandedVm, err = virtClient.ExpandSpec(vm.Namespace).ForVirtualMachine(vm)
+		if err != nil {
+			return fmt.Errorf("error expanding VirtualMachine %s: %v", vm.Name, err)
+		}
+	} else if len(args) == 1 {
```

Comment:
Make sure that --file is not used when providing the name of a VM? (if arg count > 0 then file flag should be unchanged). It could be implemented in the `Args` field of the command. Similar to here: https://github.com/spf13/cobra/blob/main/user_guide.md#positional-and-custom-arguments